iModelCore/iModelPlatform/DgnCore/DgnElements.cpp

```diff
@@ -511,6 +511,12 @@ DgnElementCPtr DgnElements::InsertElement(DgnElementR element, DgnDbStatus* outS
 
     DgnDbStatus ALLOW_NULL_OUTPUT(stat,outStat);
 
+    if ( !element.m_federationGuid.IsValid())

```

Comment:
@pmconne, when you said we don't check if a guid is valid before inserting, was this what you had in mind when suggesting we should check? Based on what I found while looking around, the `fromString()` function does seem to check that the guid is valid or else it fails, so I don't think we can insert a bad guid that way.

Reply:
I was referring to [this](https://github.com/iTwin/imodel-native/blob/c11f767346b77bdae40811793599a8be758dfc00/iModelJsNodeAddon/JsInteropDgnDb.cpp#L375)

```
        // if no federationGuid was supplied, create one for the element before we add it.
        if (!inJson.isStringMember(json_federationGuid()))
            el->SetFederationGuid(BeGuid(true));
```

If the TypeScript caller does, e.g., `insertElement({ ...props, federationGuid: "not a valid guid" })`, then the code above will not assign a new, valid guid to the element.

Reply:
going to make a separate PR for this then since this change would not be needed to BP to imodel02, but the rest of these changes would